sql/plan/load_data.go

```diff
@@ -0,0 +1,292 @@
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+)
+
+type LoadData struct {
+	Local              bool
+	File               string
+	Destination        sql.Node
+	ColumnNames        []string
+	ResponsePacketSent bool
+	Fields             *sqlparser.Fields
+	Lines              *sqlparser.Lines
+	IgnoreNum          int64
+}
+
+const (
+	Tmpfiledir  = "/tmp/"
+	TmpfileName = ".LOADFILE"
+)
+
+var (
+	fieldsTerminatedByDelim = "\t"
+	fieldsEnclosedByDelim   = ""
+	fieldsOptionallyDelim   = false
+	fieldsEscapedByDelim    = "\\"
+	linesTerminatedByDelim  = "\n"
+	linesStartingByDelim    = ""
+)
+
+func (l *LoadData) Resolved() bool {
+	return l.Destination.Resolved()
+}
+
+func (l *LoadData) String() string {
+	pr := sql.NewTreePrinter()
+
+	_ = pr.WriteNode("LOAD DATA")
+	_ = pr.WriteChildren(l.Destination.String())
+	return pr.String()
+}
+
+func (l *LoadData) Schema() sql.Schema {
+	return l.Destination.Schema()
+}
+
+func (l *LoadData) Children() []sql.Node {
+	return []sql.Node{l.Destination}
+}
+
+func (l *LoadData) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	// TODO: Add the security variables for mysql
+
+	// Start the parsing by grabbing all the config variables.
+	err := l.updateParsingConsts()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: Add tmpdir setting for mysql
+	var fileName = l.File
+	if l.Local {
+		fileName = Tmpfiledir + TmpfileName
+	}
+
+	file, err := os.Open(fileName)
+	if err != nil {
+		return nil, err
+	}
+	if l.Local {
+		defer os.Remove(fileName)
+	} else {
+		defer file.Close()
+	}
+
+	scanner := bufio.NewScanner(file)
+	parseLines(scanner)
+
+	var values [][]sql.Expression
+	scanner = bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if l.IgnoreNum <= 0 {
+			exprs, err := parseFields(line)
+
+			if err != nil {
+				return nil, err
+			}
+
+			// Line was skipped
+			if exprs == nil {
+				continue
+			}
+
+			// Match input columns with the amount of columns provided in the text.
+			// Append nils to the parsed fields if they are less than the input columns.
+			// TODO: Match schema with column order
+			colDiff := len(l.Schema()) - len(exprs)
+
+			// append NULLS for the rest of the fields
+			exprs = addNullsToValues(exprs, colDiff)
+
+			values = append(values, exprs)
+		} else {
+			l.IgnoreNum--
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	newValue := NewValues(values)
+
+	return newValue.RowIter(ctx, row)
+}
+
+func addNullsToValues(exprs []sql.Expression, diff int) []sql.Expression {
+	for i := diff; i > 0; i-- {
+		exprs = append(exprs, expression.NewLiteral(nil, sql.Null))
+	}
+
+	return exprs
+}
+
+// updateParsingConsts parses the LoadData object to update the 5 constants defined at top of the file.
+func (l *LoadData) updateParsingConsts() error {
+	if l.Lines != nil {
+		ll := l.Lines
+		if ll.StartingBy != nil {
+			linesStartingByDelim = string(ll.StartingBy.Val)
```

Comment:
This is the wrong approach. These vary by statement, so make them fields on the struct, not global vars